samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -85,7 +71,27 @@ public void testStreamProcessor() {
     // Note: createTopics needs to be called before creating a StreamProcessor. Otherwise it fails with a
     // TopicExistsException since StreamProcessor auto-creates them.
     createTopics(inputTopic, outputTopic);
-    final StreamProcessor processor = new StreamProcessor(new MapConfig(configs), new HashMap<>(), IdentityStreamTask::new, listener);
+    final StreamProcessor processor = new StreamProcessor(
+        "1",
+        new MapConfig(configs),
+        new HashMap<>(),
+        IdentityStreamTask::new,
+        new StreamProcessorLifeCycleAware() {
```

Comment:
I am not a big fan of mocks at this time. I think mocks are useful when the APIs are relatively stable/mature. Given how frequently we are shuffling classes around, I prefer not to rely on mocks so much.

One of the reasons why I had to refactor LocalApplicationRunner this way is that it was very difficult to change the mocked behavior (spy,mock, capture etc) in the test class. I will refrain from using further mocks until we can stabilize the components.